### PR TITLE
passenger: fix ruby.h header include path for new xcode

### PR DESCRIPTION
**Pull request by CamJN**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by CamJN** [comment]
So the latest build failed because there is no such file or directory: `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/include/ruby-2.3.0` in Mojave with the env:

```
macOS: 10.14.6-x86_64
12:01:34 CLT: 11.2.0.0.1.1571444319
12:01:34 Xcode: 11.2.1
12:01:34 CLT headers: 11.2.0.0.1.1571444319
```

From my research, in order to get the correct ruby headers for 10.14, you must use the CLT sdk for 10.14 at `MacOS::CLT.sdk_path`. Replacing the `ENV.delete("SDKROOT")` with `ENV["SDKROOT"] = MacOS::CLT.sdk_path` in the passenger formula worked locally on a mojave box with the latest Xcode and the CLT tools installed but without the extra transitional header package installed (ie `/Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg` is NOT installed). However when I tested it in the homebrew CI it had the same result as the most recent build where extconf still looked for the headers in the 10.15 path.

I notice that the `HOMEBREW_SDKROOT` is set to `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk` which seems wrong, should I override that too, or instead, or is there a more elegant approach?

**Comment by CamJN** [comment]
Going to try something like https://github.com/Homebrew/homebrew-core/pull/47035

**Comment by Bo98** [comment]
Apologies for the delayed response here. Any fix was impossible until today as the 10.14 SDK on the Mojave nodes were broken. #47035 was the first attempt on trying it on Ruby headers.

Indeed I recommend the approach I put in place in #47035. You will probably want a `>= :mojave` and a CLT installed check. But you can wait for the build to finish to see how necessary the Mojave check is - you might need to keep the ENV.delete for High Sierra. We'll find out.

File: Formula/passenger.rb [hunk 1]
```diff
@@ -18,9 +19,13 @@ class Passenger < Formula
...
 
+  patch do
+    url "https://github.com/phusion/passenger/commit/e512231f.patch?full_index=1"
+    sha256 "9f39f5c1c8b68516f7bac0ba07921144a5de30b6a72ef2423ea83a77d512bea8"
+  end
+
   def install
-    # https://github.com/Homebrew/homebrew-core/pull/1046
-    ENV.delete("SDKROOT")
+    ENV["SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version)
```
Comment (Bo98) [inline Formula/passenger.rb]:
```suggestion
    if MacOS.version >= :mojave && MacOS::CLT.installed?
      ENV["SDKROOT"] = MacOS::CLT.sdk_path(MacOS.version)
    else
      ENV.delete("SDKROOT")
    end
```

**Review by Bo98 — CHANGES_REQUESTED** [review]

File: Formula/passenger.rb [hunk 2]
```diff
@@ -18,9 +19,13 @@ class Passenger < Formula
   depends_on "openssl@1.1"
   depends_on "pcre"
 
+  patch do
+    url "https://github.com/phusion/passenger/commit/e512231f.patch?full_index=1"
+    sha256 "9f39f5c1c8b68516f7bac0ba07921144a5de30b6a72ef2423ea83a77d512bea8"
+  end
```
Comment (Bo98) [inline Formula/passenger.rb]: What's this and the `passenger_temp_path` about? Could use a comment.

Comment (CamJN) [reply Formula/passenger.rb]: True, it's to address a failure seen here: https://github.com/Homebrew/homebrew-core/pull/45685#issuecomment-550307627 (sandbox violation) I started this PR to test the ruby headers fix, but when that wasn't working I just started using it to test the temp path issue instead then circled back to the headers thing now that the temp path part is working.

Comment (CamJN) [reply Formula/passenger.rb]: Added comment

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request closed without merging**